appserver/web/web-core/src/main/java/org/apache/catalina/authenticator/FormAuthenticator.java

```diff
@@ -148,6 +153,12 @@ public boolean authenticate(HttpRequest request, HttpResponse response, LoginCon
         
         // Is this the action request from the login page?
         boolean loginAction = requestURI.startsWith(contextPath) && requestURI.endsWith(FORM_ACTION);
+        if (loginAction) {
+            if (!isPermittedHttpMethod(hreq.getMethod())) {
+                forwardToErrorPage(request, response, config);
```

Comment:
I think you maybe just want to sent a `deny` here. The error page is often tailored for wrong input (i.e. credentials did not validate).

Reply:
Changed to deny.